### PR TITLE
Add patch endpoint to assessment api

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
Required for amending the status values for application summary display in caab-ui

### CODE REVIEW

File: assessment-api/open-api-specification.yml [hunk 1]
```diff
@@ -70,6 +70,41 @@ paths:
           description: 'Not found'
         '500':
           description: 'Internal server error'
+    patch:
+      tags:
+        - assessments
+      summary: 'update assessment'
+      operationId: 'updateAssessment'
+      requestBody:
+        description: Create a application
```
Comment (porritta) [inline assessment-api/open-api-specification.yml:79]: wrong description

File: assessment-api/open-api-specification.yml [hunk 2]
```diff
@@ -99,10 +142,24 @@ components:
           items:
             $ref: "#/components/schemas/assessmentDetail"
     assessmentDetail:
+      allOf:
+        - $ref: "#/components/schemas/baseAssessmentDetail"
+        - type: 'object'
+          properties:
+            id:
```
Comment (porritta) [inline assessment-api/open-api-specification.yml:149]: Seems strange that id is in the extended assessmentDetail and not in the base object

**Review by porritta — CHANGES_REQUESTED** [review]
review complete

File: assessment-api/open-api-specification.yml (hunk 2 above)
Comment (PhilDigitalJustice) [reply assessment-api/open-api-specification.yml:149]: You don't want to be able to amend the id as part of the patch

File: assessment-api/open-api-specification.yml (hunk 1 above)
Comment (PhilDigitalJustice) [reply assessment-api/open-api-specification.yml:79]: amended

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**